packages/workspace/test/workspace/multi_env/routers.test.ts

```diff
@@ -828,6 +855,46 @@ describe('isolated routing', () => {
       path: ['test', 'path'],
     })
   })
+
+  it('should route a common modification diff to comon and revert the change in'
+   + 'secondary envs for nested elements without parents in the env', async () => {
+    const fieldID = commonOnlyFieldObjectID.createNestedID('field', 'ofdreams')
+    const specificChange: DetailedChange = {
+      action: 'modify',
+      data: { before: 'if you will build it', after: 'they will come' },
+      id: fieldID.createNestedID('catchphrase'),
+    }
+    const beforeField = commonCommonOnlyFieldObject.fields.ofdreams
+    const afterField = _.clone(beforeField)
```

Comment:
cloning a field creates an invalid field (it clones the pointer to the parent type but the parent type does not point back to the cloned field).
you have to create a new object type with the new field
```suggestion
    const newObject = new ObjectType({
      ... commonCommonOnlyFieldObject,
      fields: {
        ofdreams: { ...commonCommonOnlyFieldObject.fields.ofdreams, anootations: { catchphrase: ... } },
      },
    })
    const afterField = newObject.fields.ofdreams
```